common/app/common/NewNavigation.scala

```diff
@@ -498,19 +498,19 @@ object NewNavigation {
 
     def getSectionOrTagId(page: Page) = {
       val tags = Navigation.getTagsFromPage(page)
-      val commonKeywords = tagPages.intersect(tags.keywordIds)
+      val commonKeywords = tags.keywordIds.intersect(tagPages)
```

Comment:
Not fussy! Its a really good point. I have made the change. I spent some time looking, but I couldn't find a simpler way to do it than using `sortWith` like you showed me.